src/test/java/nextstep/subway/line/application/LineServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package nextstep.subway.line.application;
+
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.line.domain.LineRepository;
+import nextstep.subway.line.dto.LineRequest;
+import nextstep.subway.line.dto.LineResponse;
+import nextstep.subway.station.application.StationService;
+import nextstep.subway.station.dto.StationResponse;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static nextstep.subway.line.domain.LineFixture.신이호선;
+import static nextstep.subway.line.domain.LineFixture.이호선;
+import static nextstep.subway.station.StationFixture.당산역;
+import static nextstep.subway.station.StationFixture.합정역;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@Service
```

Comment:
`@Service` 는 bean을 만들기 위한 메소드에요.
단위 테스트에서는 bean을 사용하지 않는게 원칙입니다.

Junit5 에서 Mockito를 사용하시려면 이렇게 교체해주세요.
```suggestion
@ExtendWith(MockitoExtension.class)
```

저는 mockito를 설명할 때 과학 실험을 예시로 들고 있는데요,
어렸을 적 과학 시간에 배운 실험군과 대조군을 기억하시나요?

대조군은 원본 그대로를 말하고 실험군은 다른 조건은 그대로 둔 채 실험 조건만 변경하여 적용해 둔 집단을 말합니다.
여기서의 실험군은 테스트 대상인 `LineService` 의 메소드가 되겠지요.

그리고 이 테스트 클래스에서 검증하고자 하는건 "서비스 메소드에게 기대한 로직을 잘 실행하는가" 입니다.
외부 의존성이 변하던 말던 이 로직 자체는 주어진 목적을 수행해야 한다는 말입니다.

그런 이유로 bean으로 주입한 외부 의존성은 값이 고정되어 있다고 가정하고 진행합니다.
현재 테스트 대상의 로직만을 집중 검증하는게 목적이기 때문이에요.
따라서 외부 의존성은 mocking하여 값을 고정시키고 대상 메소드의 로직만 검증하는 것입니다.

외부 의존성이 변하는 것까지 고려하고 싶다면 bean을 주입하고 통합 테스트를 작성하시면 됩니다.
지금은 단위 테스트에 대해서만 다루고 있으니 이건 별도로 공부해보시면 좋겠네요.

Reply:
명쾌한 비유로 설명해주셔서 감사합니다!
위에 추가로 알려주신 내용도 보고 설명해주신 부분들도 보면서 어떤 개념인지 왜 그렇게 해야하는지 등을 조금 더 이해할 수 있게 되었네요 👍

위에 피드백 주신 내용도 반영했습니다.